go/obsclient/authclient.go

```diff
@@ -0,0 +1,78 @@
+package obsclient
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+)
+
+// AuthObsClient extends the functionality of the ObsClient for all methods that require encryption when communicating with the enclave
+type AuthObsClient struct {
+	ObsClient
```

Comment:
Extending ObsClient like this lets us call the non-authenticated methods on the auth client, it makes the struct initialization look a bit weird though, probably fine because it's hidden behind constructor function but open to ideas.